modules/kafka/kafka_test.go

```diff
@@ -97,16 +100,258 @@ func TestKafka_invalidVersion(t *testing.T) {
+		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "INTERNAL",
+				Ip:   "kafka",
+				Port: "9093",
+			},
+		}),
+	)
+
+	if err == nil {
+		t.Fatalf("expected to fail due to reserved listener port duplication")
+	}
+
+	_, err = kafka.RunContainer(ctx,
+		kafka.WithClusterID("test-cluster"),
+		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "INTERNAL",
+				Ip:   "kafka",
+				Port: "9094",
+			},
+		}),
+	)
+
+	if err == nil {
+		t.Fatalf("expected to fail due to reserved listener port duplication")
+	}
+
+	_, err = kafka.RunContainer(ctx,
+		kafka.WithClusterID("test-cluster"),
+		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "  cOnTrOller   ",
+				Ip:   "kafka",
+				Port: "9092",
+			},
+		}),
+	)
+
+	if err == nil {
+		t.Fatalf("expected to fail due to reserved listener name duplication")
+	}
+
+	_, err = kafka.RunContainer(ctx,
+		kafka.WithClusterID("test-cluster"),
+		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "external",
+				Ip:   "kafka",
+				Port: "9092",
+			},
+		}),
+	)
+
+	if err == nil {
+		t.Fatalf("expected to fail due to reserved listener name duplication")
+	}
+
+	_, err = kafka.RunContainer(ctx,
+		kafka.WithClusterID("test-cluster"),
+		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "test",
+				Ip:   "kafka",
+				Port: "9092",
+			},
+			{
+				Name: "test2",
+				Ip:   "kafka",
+				Port: "9092",
+			},
+		}),
+	)
+
+	if err == nil {
+		t.Fatalf("expected to fail due to port duplication")
+	}
+
+	_, err = kafka.RunContainer(ctx,
+		kafka.WithClusterID("test-cluster"),
+		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "test",
+				Ip:   "kafka",
+				Port: "9092",
+			},
+			{
+				Name: "test",
+				Ip:   "kafka",
+				Port: "9095",
+			},
+		}),
+	)
 
+	if err == nil {
+		t.Fatalf("expected to fail due to name duplication")
+	}
+}
+
+func initKafkaTest(ctx context.Context, network string, brokers string, input string, output string) (testcontainers.Container, error) {
+	req := testcontainers.ContainerRequest{
+		FromDockerfile: testcontainers.FromDockerfile{
+			Context:       "./testdata",
+			Dockerfile:    "Dockerfile",
+			PrintBuildLog: true,
+			KeepImage:     true,
+		},
+		WaitingFor: wait.ForLog("start consuming events"),
+		Env: map[string]string{
+			"KAFKA_BROKERS":   brokers,
+			"KAFKA_TOPIC_IN":  input,
+			"KAFKA_TOPIC_OUT": output,
+		},
+		Networks: []string{network},
+	}
+
+	return testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+}
```

Comment:
This can be reduced by using kcat. See https://github.com/testcontainers/testcontainers-java/blob/main/modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java#L188-L216

Reply:
will take a look